### PR TITLE
[titanium] update types for Titanium 9.0.0

### PR DESCRIPTION
**Pull request by janvennemann**
Updates the typings for non-npm package Titanium to v9.0. There were a couple of substantial changes made to the generator of these typing, resulting in a lot of changes to the definitions. These changes are described in https://github.com/appcelerator/docs-devkit/pull/25.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: Generated from https://github.com/appcelerator/titanium_mobile/tree/9_0_X/apidoc
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/titanium/tsconfig.json [hunk 1]
```diff
@@ -2,8 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "dom",
-            "es6"
+            "es5"
```
Comment (amcasey) [inline types/titanium/tsconfig.json]: I don't usually see this go backwards.  Intentional?

**Comment by amcasey** [comment]
> //                 Mathias Lorenzen <https://github.com/ffMathy>

An owner was dropped.  Intentional?  If so, do you want to tag the new owner on the PR instead?

---
Refers to: types/titanium/index.d.ts:5 in 05660f6. [](commit_id = 05660f61bf5ca9cd043355405e54d7603e7e6cf6, deletion_comment = True)

**Review by amcasey — APPROVED** [review]
There's no way I can review a diff this large, but I have a couple of minor questions.

**Comment by ffMathy** [comment]
Why was I dropped as owner?

**Review by ffMathy — CHANGES_REQUESTED** [review]
Why drop me as owner? 🙁

File: types/titanium/tsconfig.json (hunk 1 above)
Comment (janvennemann) [reply types/titanium/tsconfig.json]: Yes, these were wrong before. `dom` was only used to get typings for `console`, which are now included in our typings. Also the typings do not use any ES6 features yet, so we changed that as well and only include `es5`.

**Comment by janvennemann** [comment]
@ffMathy sorry man, it appears you didn't adjust the header in our TypeScript generator. I'll add you back and push an update!

**Comment by ffMathy** [comment]
@janvenneman thanks!

**Comment by amcasey** [comment]
LGTM, but it would be nice to have confirmation from @ffMathy.

**Comment by ffMathy** [comment]
All good :+1:

**Pull request merged**